nautobot/project-static/js/forms.js

```diff
@@ -127,6 +129,88 @@ function initializeSlugField(context){
     }
 }
 
+function initializePositionField(context){
+    this_context = $(context);
+    var position_field = this_context.find('#id_position');
+    if (position_field.length != 0) {
+        var source_arr = position_field.attr('source').split(" ");
+        var length = position_field.attr('maxlength');
+        if (position_field.val()) {
+            position_field.attr('_changed', true);
+        } else {
+            position_field.attr('_changed', false);
+        }
+        position_field.change(function() {
+            if (!position_field.val()) {
+                $(this).attr('_changed', false);
+            } else {
+                $(this).attr('_changed', true);
+            }
+        });
+        function repopulate() {
+            let str = "";
+            for (source_str of source_arr) {
+                if (str != "") {
+                    str += " ";
+                }
+                let source = $('#id_' + source_str);
+                str += source.val();
+            }
+            position_field.val(str, (length ? length : 100));
+        };
+        for (source_str of source_arr) {
+            let source = $('#id_' + source_str);
+            source.on('keyup change', function() {
+                if (position_field && position_field.attr('_changed')=="false") {
+                    repopulate();
+                }
+            });
+        }
+        this_context.find('button.reslugify').click(repopulate);
+    }
+}
+
+function initializePositionPatternField(context){
+    this_context = $(context);
+    var position_field = this_context.find('#id_position_pattern');
```

Comment:
Is this the only difference between these two functions? Can we consolidate logic into a common helper function if so?